### DISCUSSION TITLE
Concurrency, mapping dynamic closures.

### DISCUSSION THREAD

**Discussion by ovp87** [laravel/framework#53134] [Ideas]
Is there a convenient way of mapping the output, if you have a dynamic amount of closures?

```
$result = Concurrency::run([
    // some dynamic way of generating an unknown number of these
    'task1' => function () {
        sleep(5);
        return 'Result of task 1';
    },
    'task2' => function () {
        sleep(4);
        return 'Result of task 2';
    },
]);
```

`$result = Concurrency::run($data);`

Ideally, id like this to respect the keys that was passed in the input:
`['task1' => 'Result of task 1', 'task2' => 'Result of task2']`

But it that doesn't seem to be the case, the output is not keyed to the keys of the array input, unless i'm missing something?

Edit: added a PR here:
https://github.com/laravel/framework/pull/53135

**Comment by murilolivorato** [comment]
```
function generateTasks($numTasks) {
    $tasks = [];
    for ($i = 1; $i <= $numTasks; $i++) {
        $tasks["task$i"] = function () use ($i) {
            sleep(6 - $i); // Example sleep time
            return "Result of task $i";
        };
    }
    return $tasks;
}
```

$tasks = generateTasks(5); // Example with 5 tasks
$results = Concurrency::run($tasks);

// Map results back to original keys
$mappedResults = array_combine(array_keys($tasks), $results);

print_r($mappedResults);

**Reply by ovp87** [reply]
Sure, you could obviously do this, just a much better looking experience when its done out of the box.
